From 3f3ad56b5237fe558fccff80046bc79c7a1ded0c Mon Sep 17 00:00:00 2001
From: Vlad Yasevich <vyasevic@redhat.com>
Date: Fri, 29 Aug 2014 19:01:23 -0400
Subject: [net] bridge: fix the unbalanced promiscuous count when add_if failed

Message-id: <1409338885-30414-12-git-send-email-vyasevic@redhat.com>
Patchwork-id: 89908
O-Subject: [RHEL7 PATCH 11/13] bridge: fix the unbalanced promiscuous count when add_if failed
Bugzilla: 896669
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>

From: wangweidong <wangweidong1@huawei.com>

As commit 2796d0c648c94 ("bridge: Automatically manage port
promiscuous mode."), make the add_if use dev_set_allmulti
instead of dev_set_promiscuous, so when add_if failed, we
should do dev_set_allmulti(dev, -1).

Signed-off-by: Wang Weidong <wangweidong1@huawei.com>
Reviewed-by: Amos Kong <akong@redhat.com>
Acked-by: Vlad Yasevich <vyasevic@redhat.com>
Acked-by: Stephen Hemminger <stephen@networkplumber.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 019ee792d7861fde33e6c2331a8b32a2b975e865)
Signed-off-by: Vladislav Yasevich <vyasevic@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/bridge/br_if.c b/net/bridge/br_if.c
index dddb8bf..b6396d3 100644
--- a/net/bridge/br_if.c
+++ b/net/bridge/br_if.c
@@ -524,7 +524,7 @@ err2:
  kobject_put(&p->kobj);
  p = NULL; /* kobject_put frees */
 err1:
- dev_set_promiscuity(dev, -1);
+ dev_set_allmulti(dev, -1);
 put_back:
  dev_put(dev);
  kfree(p);
-- 
1.7.1